### PR TITLE
build `blueprint-compiler` from source

### PR DESCRIPTION
**Pull request by soumyaDghosh**
Changes in this PR

1. Building `blueprint-compiler` from source
2. removing the patch

@fsobolev do you think that the patch worth maintaining?

### CODE REVIEW

**Comment by nlogozzo** [comment]
> do you think that the patch worth maintaining?

What was the purpose of the patch in the first place? Wasn't it because the icon didn't show without it?

**Comment by soumyaDghosh** [comment]
> What was the purpose of the patch in the first place? Wasn't it because the icon didn't show without it?

The patch is there to show the symbolic icon in the gnome activities icon.

**Comment by nlogozzo** [comment]
> > What was the purpose of the patch in the first place? Wasn't it because the icon didn't show without it?
> 
> The patch is there to show the symbolic icon in the gnome activities icon.

So why would we get rid of that...

**Comment by soumyaDghosh** [comment]
> > > What was the purpose of the patch in the first place? Wasn't it because the icon didn't show without it?
> > 
> > 
> > The patch is there to show the symbolic icon in the gnome activities icon.
> 
> So why would we get rid of that...

This much change just for a symbolic icon! It's not like the app doesn't show any icon, it just shows the black and white version of the app's original icon

**Comment by nlogozzo** [comment]
> This much change just for a symbolic icon!

I don't mind maintaining it to show the correct icon. It's not like the desktop file changes often where we need to fix the patch. Just when we move things around, but I update the patch.

Please restore it back, I'll maintain it dw

File: snap/denaro.patch
Comment (nlogozzo) [inline snap/denaro.patch:1]: Restore please

File: .github/workflows/snap.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 on:
   push:
-    branches: [ "main" ]
+    branches: [ "main", "snap" ]
```
Comment (nlogozzo) [inline .github/workflows/snap.yml:3]: remove `"snap"` as testing is over

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -1,32 +1,46 @@
...
+    #build-snaps: # when available
+    #- blueprint-compiler
+    build-packages: 
       - wget
     override-pull: |
       craftctl default
       wget https://dot.net/v1/dotnet-install.sh -O dotnet-install.sh
       chmod +x ./dotnet-install.sh
       ./dotnet-install.sh --channel 8.0
-      patch -p1 < $CRAFT_PROJECT_DIR/snap/denaro.patch
```
Comment (nlogozzo) [inline snap/snapcraft.yaml:26]: restore

**Review by nlogozzo — CHANGES_REQUESTED** [review]

**Review by nlogozzo — APPROVED** [review]

**Comment by nlogozzo** [comment]
Could you do the same PR for Tagger please?

**Comment by soumyaDghosh** [comment]
Sure.. just a moment, working on `snapcraft`

**Pull request merged**